tests for JBJCA-747, JBJCA-825, JBJCA-826 + minor jca test changes

simkam opened the pull request:
tests for:
https://issues.jboss.org/browse/JBJCA-747 LazyAssociatableConnectionManager/NoTx
https://issues.jboss.org/browse/JBJCA-825 LazyAssociatableConnectionManager/Tx
https://issues.jboss.org/browse/JBJCA-826 LazyEnlistableConnectionManager

More or less straight port from the IronJacamar test suite from package org.jboss.jca.deployers.test.unit.lazy. 

Included only basic tests to ensure that integration works - tests without transaction, with local transaction and with xa transaction. Omitted tests with multiple threads and with interleaving. 
- in comparison to IJ TS there are no significant changes - changed logging messages and log level

Also added some additional tests with RA that implements DissociatableManagedConnection, LazyEnlistableManagedConnection but both is disabled in configuration. 
- added basic test for JCA 1.7 RA deployment, which isn't related to lazy connection manager, but I noticed that it's missing

simkam committed:
JCA 1.7 RA deployment test case

simkam committed:
tests for JBJCA-747, JBJCA-825, JBJCA-826

https://issues.jboss.org/browse/JBJCA-747 LazyAssociatableConnectionManager/NoTx
https://issues.jboss.org/browse/JBJCA-825 LazyAssociatableConnectionManager/Tx
https://issues.jboss.org/browse/JBJCA-826 LazyEnlistableConnectionManager

kabir commented:
@jesperpedersen and/or @maeste please approve

jesperpedersen commented:
* Missing license headers
* Looks like more or less as a straight port from the IronJacamar test suite => copyright needs to be maintained
* Missing description on the PR of what (if anything) has been changed, and what is tested
* Missing PR squash

simkam commented:
* added license headers 
* added author from IJ TS, is it enough? or how can I maintain copyright?
* PR description updated
* I didn't squash commits, because they aren't really related, but I can do it if you wish

jesperpedersen commented:
@kabir LGTM

simkam commented:
rebased

bstansberry commented:
Please get rid of the EAP7-177 reference from the commit message. That's an internal, not publicly visible planning item, not relevant to WildFly.

Otherwise, ready to merge.
